src/EditorFeatures/Test/CodeFixes/CodeFixServiceTests.cs

```diff
@@ -924,5 +937,224 @@ internal sealed class AdditionalFileFixerWithoutDocumentKindsAndExtensions : Abs
+            //     snapshots have the same number of lines.
+
+            // We expect de-prioritization of analyzer from 'Normal' to 'Low' bucket only if following conditions are met:
+            //  1. There are no cached diagnostics from background analysis, i.e. 'testWithCachedDiagnostics == false'.
+            //  2. We have an expensive analyzer that registers SymbolStart/End or SemanticModel actions, both of which have a broad analysis scope.
+            //  3. Either of the below is true:
+            //     a. We do not have an analyzer diagnostic reported in the prior document snapshot on the edited line OR
+            //     b. Number of lines in the prior document snapshot differs from number of lines in the current document snapshot.
+            var boolExpectDeprioritization = !testWithCachedDiagnostics &&
+                (actionKind is DeprioritizedAnalyzer.ActionKind.SymbolStartEnd or DeprioritizedAnalyzer.ActionKind.SemanticModel) &&
+                (!diagnosticOnFixLineInPriorSnapshot || addNewLineWithEdit);
+
+            var priorSnapshotFixLine = diagnosticOnFixLineInPriorSnapshot ? "int x1 = 0;" : "System.Console.WriteLine();";
+            var code = $@"
+#pragma warning disable CS0219
+class C
+{{
+    void M()
+    {{
+        {priorSnapshotFixLine}
+    }}
+}}";
+
+            var codeFix = new FixerForDeprioritizedAnalyzer();
+            var analyzer = new DeprioritizedAnalyzer(actionKind);
+            var analyzerReference = new MockAnalyzerReference(codeFix, ImmutableArray.Create<DiagnosticAnalyzer>(analyzer));
+
+            var tuple = ServiceSetup(codeFix, code: code);
+            using var workspace = tuple.workspace;
+            GetDocumentAndExtensionManager(tuple.analyzerService, workspace, out var document,
+                out var extensionManager, out var diagnosticIncrementalAnalyzer, analyzerReference);
+
+            var sourceDocument = (Document)document;
+            var root = await sourceDocument.GetSyntaxRootAsync();
+            var testSpan = diagnosticOnFixLineInPriorSnapshot
+                ? root!.DescendantNodes().OfType<CodeAnalysis.CSharp.Syntax.VariableDeclarationSyntax>().First().Span
+                : root!.DescendantNodes().OfType<CodeAnalysis.CSharp.Syntax.InvocationExpressionSyntax>().First().Span;
+
+            // Trigger background analysis to ensure analyzer diagnostics are computed and cached. 
+            // We enable full solution analysis so the 'AnalyzeDocumentAsync' doesn't skip analysis based on whether the document is active/open.
+            workspace.GlobalOptions.SetGlobalOption(SolutionCrawlerOptionsStorage.BackgroundAnalysisScopeOption, LanguageNames.CSharp, BackgroundAnalysisScope.FullSolution);
+            await diagnosticIncrementalAnalyzer.AnalyzeDocumentAsync(sourceDocument, bodyOpt: null!, InvocationReasons.DocumentChanged, CancellationToken.None);
```

Comment:
why `null!` if it's explicitly allowed to be null?

Reply:
This is an unused parameter, but seems to exist because it is an override.